### PR TITLE
ArgumentError: comparison of Carto::Visualization with Carto::Visualization failed

### LINKED ISSUE #4104: ArgumentError: comparison of Carto::Visualization with Carto::Visualization failed
https://rollbar.com/vizzuality/CartoDB/items/7743/
```
ArgumentError: comparison of Carto::Visualization with Carto::Visualization failed
1
File "/home/ubuntu/www/production.cartodb.com/releases/20150617144303/app/queries/carto/offdatabase_query_adapter.rb", line 55 in sort
2
File "/home/ubuntu/www/production.cartodb.com/releases/20150617144303/app/queries/carto/offdatabase_query_adapter.rb", line 55 in block in get_results
3
File "/home/ubuntu/www/production.cartodb.com/releases/20150617144303/app/queries/carto/offdatabase_query_adapter.rb", line 52 in each
4
File "/home/ubuntu/www/production.cartodb.com/releases/20150617144303/app/queries/carto/offdatabase_query_adapter.rb", line 52 in get_results
5
File "/home/ubuntu/www/production.cartodb.com/releases/20150617144303/app/queries/carto/offdatabase_query_adapter.rb", line 45 in results
6
File "/home/ubuntu/www/production.cartodb.com/releases/20150617144303/app/queries/carto/offdatabase_query_adapter.rb", line 33 in map
7
File "/home/ubuntu/www/production.cartodb.com/releases/20150617144303/app/controllers/carto/api/visualizations_controller.rb", line 37 in index
... 6 non-project frames
14
File "/home/ubuntu/www/production.cartodb.com/releases/20150617144303/app/controllers/application_controller.rb", line 54 in wrap_in_profiler
... 67 non-project frames
```

happens in the dashboard when sorting by size, seems to be a corner case (nil size?)

cc/ @juanignaciosl @Kartones

### PR DESCRIPTION
**Pull request by Kartones**
Closes #4104

### CODE REVIEW

**Comment by Kartones** [comment]
@rafatower quick CR please

File: app/models/carto/visualization.rb [hunk 1]
```diff
@@ -43,6 +43,10 @@ class Carto::Visualization < ActiveRecord::Base
 
   belongs_to :map
 
+  def ==(other_visualization)
```
Comment (Kartones) [inline app/models/carto/visualization.rb]: Not needed but I'd like to start being able to compare equality using objects and not the `.id` everywhere so...

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4671/))

**Comment by rafatower** [comment]
:+1:

**Pull request merged**